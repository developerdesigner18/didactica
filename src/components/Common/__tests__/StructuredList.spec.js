import { createLocalVue, mount } from "@vue/test-utils";

import draggable from "vuedraggable";
import LxpStructuredList from "../StructuredList.vue";
import { BTooltip } from "bootstrap-vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-structured-list", LxpStructuredList);
localVue.component("draggable", draggable);
localVue.component("b-tooltip", BTooltip);

describe("StructuredList.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });
  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should contain a list with draggable icon", async () => {
      wrapper = await getWrapper();
      const listIcon = wrapper.find(".lxp-sl__list--item__handler");
      expect(listIcon.exists()).toBe(true);
    });
  });
  describe("Events", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should emit an event when an item is dragged", async () => {
      wrapper = await getWrapper();
      const draggableItem = wrapper.find(".lxp-sl__list--item__handler");
      draggableItem.trigger("dragstart");
      draggableItem.trigger("dragend");
      expect(wrapper.exists("item-dragged")).toBeTruthy();
    });
  });

  describe("Data", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });

    it("should have correct data when created", async () => {
      wrapper = await getWrapper();
      const items = wrapper.vm.items;
      expect(items).toEqual([
        { name: "First", order: 1, fixed: false },
        { name: "Second", order: 2, fixed: false },
        { name: "Third", order: 3, fixed: false }
      ]);
    });
  });
});

function getWrapper() {
  return mount(LxpStructuredList, {
    propsData: {
      items: [
        {
          name: "First",
          order: 1,
          fixed: false
        },
        {
          name: "Second",
          order: 2,
          fixed: false
        },
        {
          name: "Third",
          order: 3,
          fixed: false
        }
      ]
    },
    localVue
  });
}
