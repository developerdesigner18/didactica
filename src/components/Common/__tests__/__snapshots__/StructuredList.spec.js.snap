// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StructuredList.vue render when created  1`] = `
<div
  class="lxp-structured-list"
>
  <div>
    <transition-group-stub
      name="list"
    >
      <div
        class="lxp-sl__list--item"
      >
        <span>
          <div
            class="lxp-sl__list--item__handler"
          >
            <img
              alt="drag"
              id="imgHandle-0"
              loading="lazy"
              src="https://files.lxp.academy.who.int/didactica/assets/images/learning-space/main/dragging-indicator.svg"
            />
             
            <!---->
             
            <span
              class="lxp-sl__list--name"
            >
              First
            </span>
          </div>
        </span>
      </div>
      <div
        class="lxp-sl__list--item"
      >
        <span>
          <div
            class="lxp-sl__list--item__handler"
          >
            <img
              alt="drag"
              id="imgHandle-1"
              loading="lazy"
              src="https://files.lxp.academy.who.int/didactica/assets/images/learning-space/main/dragging-indicator.svg"
            />
             
            <!---->
             
            <span
              class="lxp-sl__list--name"
            >
              Second
            </span>
          </div>
        </span>
      </div>
      <div
        class="lxp-sl__list--item"
      >
        <span>
          <div
            class="lxp-sl__list--item__handler"
          >
            <img
              alt="drag"
              id="imgHandle-2"
              loading="lazy"
              src="https://files.lxp.academy.who.int/didactica/assets/images/learning-space/main/dragging-indicator.svg"
            />
             
            <!---->
             
            <span
              class="lxp-sl__list--name"
            >
              Third
            </span>
          </div>
        </span>
      </div>
    </transition-group-stub>
  </div>
</div>
`;
